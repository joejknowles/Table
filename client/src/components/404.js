import React from 'react';
import res from '../resources/pages/404';

export default () => (
  <div className="not-found-message">
    { res.message }
  </div>
);
