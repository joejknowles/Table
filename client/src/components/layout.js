import React from 'react';

export default ({ children }) => (
  <div className="App">
    { children }
  </div>
);
