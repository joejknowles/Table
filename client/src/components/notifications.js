import React from 'react';
import { connect } from 'react-redux';
import { notificationSelector } from '../reducers';

export const Notifications = ({ notification }) => (
  notification ? <p>{ notification }</p> : null
);

const mapStateToProps = (state) => ({
  notification: notificationSelector(state)
});

export default connect(mapStateToProps)(Notifications);
