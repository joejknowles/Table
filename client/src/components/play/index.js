import React from 'react';
import { connect } from 'react-redux';

import Hand from './hand';
import { handSelector } from '../../reducers';
import '../../styles/Play.css';
import res from '../../resources/pages/play';

export const Play = ({ hand }) => (
  <div className="App Play">
    <div className="deck">
      { hand > 0 ?
          <Hand /> :
          <p className='no-cards-message'>
            { res.noCardsMessage }
          </p>
      }
    </div>
  </div>
);


export default connect(
  (state) => ({ hand: handSelector(state) })
)(Play);
