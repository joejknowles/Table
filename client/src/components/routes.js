import React from 'react';
import { Router, Route, browserHistory } from 'react-router';
import StartScreen from './startScreen';
import Waiting from './waiting';
import Play from './play';
import Table from './table';
import NotFound from './404';
import paths from '../shared/paths';

export default () => (
  <Router history={ browserHistory }>
    <Route path={ paths.startScreen } component={ StartScreen } />
    <Route path={ paths.play + '(/:gameCode)' } component={ Play } />
    <Route path={ paths.table } component={ Table } />
    <Route path={ paths.waiting } component={ Waiting } />
    <Route path='/*' component={ NotFound } />
  </Router>
);
