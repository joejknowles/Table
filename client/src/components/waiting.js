import React from 'react';
import { connect } from 'react-redux';

import Layout from './layout'
import { DispatchButton } from './common/buttons';
import res from '../resources/pages/waiting';
import { gameCodeSelector, playerCountSelector } from '../reducers';

export const Waiting = ({ code, playerCount }) => (
  <Layout>
    <p className="number-of-players">{ res.numOfPlayers(playerCount) }</p>
    <p className="game-code-message" >{ res.enterGameCodeMessage }</p>
    <p className="game-code" >{ code }</p>
    <DispatchButton
      res={ res.begin }
      action={ { type: 'REQUEST_BEGIN_GAME' } }
      className='btn begin-button'
    />
  </Layout>
);

const mapStateToProps = (state) => ({
  code: gameCodeSelector(state),
  playerCount: playerCountSelector(state)
});

export default connect(mapStateToProps)(Waiting);
