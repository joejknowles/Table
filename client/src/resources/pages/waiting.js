export default {
  begin: 'Begin Game',
  numOfPlayers: (num) => `${ num } players joined so far`
}
