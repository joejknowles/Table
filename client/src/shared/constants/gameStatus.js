export default {
  setup: 0,
  ongoing: 1,
  finished: 2
}
