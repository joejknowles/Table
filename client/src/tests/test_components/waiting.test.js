import { Waiting } from '../../components/waiting';

import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestUtils from 'react-addons-test-utils';

import gameStatus from '../../shared/constants/gameStatus';

const shallowRenderer = ReactTestUtils.createRenderer();

it('component renders as before with no players', () => {
  const component = shallowRenderer.render(
    <Waiting
      players={ 0 }
      gameCode={ '1234' }
    />
  );
  expect(component).toMatchSnapshot();
});

it('component renders as before with 1 player', () => {
  const component = shallowRenderer.render(
    <Waiting
      players={ 1 }
      gameCode={ '1234' }
    />
  );
  expect(component).toMatchSnapshot();
});

it('component renders as before with 2 players', () => {
  const component = shallowRenderer.render(
    <Waiting
      players={ 2 }
      gameCode={ '1234' }
    />
  );
  expect(component).toMatchSnapshot();
});
