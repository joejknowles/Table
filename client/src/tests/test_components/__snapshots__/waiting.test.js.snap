exports[`test component renders as before with 1 player 1`] = `
<Unknown>
  <p
    className="number-of-players">
    undefined players joined so far
  </p>
  <p
    className="game-code-message">
    Enter this game code on another device to play
  </p>
  <p
    className="game-code" />
  <Connect(Component)
    action={
      Object {
        "type": "REQUEST_BEGIN_GAME",
      }
    }
    className="begin-button"
    res="Begin Game" />
</Unknown>
`;

exports[`test component renders as before with 2 players 1`] = `
<Unknown>
  <p
    className="number-of-players">
    undefined players joined so far
  </p>
  <p
    className="game-code-message">
    Enter this game code on another device to play
  </p>
  <p
    className="game-code" />
  <Connect(Component)
    action={
      Object {
        "type": "REQUEST_BEGIN_GAME",
      }
    }
    className="begin-button"
    res="Begin Game" />
</Unknown>
`;

exports[`test component renders as before with no players 1`] = `
<Unknown>
  <p
    className="number-of-players">
    undefined players joined so far
  </p>
  <p
    className="game-code-message">
    Enter this game code on another device to play
  </p>
  <p
    className="game-code" />
  <Connect(Component)
    action={
      Object {
        "type": "REQUEST_BEGIN_GAME",
      }
    }
    className="begin-button"
    res="Begin Game" />
</Unknown>
`;
